derive-input/src/internal.rs

```diff
@@ -14,6 +14,35 @@ pub(crate) fn try_parse(
     args: proc_macro2::TokenStream,
     input: &syn::DeriveInput,
 ) -> syn::Result<crate::api::Table> {
-    let (table_args, column_args) = integration::spacetime_bindings_macro_input(input)?;
-    table::try_parse(args, input, &table_args, &column_args)
+    // Parse plural_name from DSL arguments first
+    let plural_name = parse_plural_name_from_args(&args)?;
+    
+    // Pass plural_name to integration for intelligent table selection
+    let (table_args, column_args) = integration::spacetime_bindings_macro_input(input, plural_name.as_ref())?;
+    
+    // Pass the parsed plural_name to avoid re-parsing
+    table::try_parse(args, input, &table_args, &column_args, plural_name)
+}
+
+// Parse plural_name from DSL arguments
+fn parse_plural_name_from_args(args: &proc_macro2::TokenStream) -> syn::Result<Option<syn::Ident>> {
+    use spacetime_bindings_macro_input::{match_meta, util::check_duplicate};
+    use syn::{meta::parser, parse::Parser};
+    use dsl::plural_name;
+    
+    let mut plural_name_value: Option<syn::Ident> = None;
+
+    parser(|meta| {
+        match_meta!(match meta {
+            plural_name => {
+                check_duplicate(&plural_name_value, &meta)?;
+                let value = meta.value()?;
+                plural_name_value = Some(value.parse()?);
+            }
+        });
+        Ok(())
+    })
+    .parse2(args.clone())?;
+
+    Ok(plural_name_value)
```

Comment:
The plural_name is requried, so please use the error check from line 27 to 30 of the derive-input/src/internal/table.rs directly here, change the return type to syn::Result<syn::Ident>.